spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -202,20 +205,27 @@ public void shutdown() throws Exception {
 
 		private final URL baseUrl;
 
+		private final int port;
+
+		private int pid;
+
 		private AppInstance(String deploymentId, int instanceNumber, ProcessBuilder builder, Path workDir, int port) throws IOException {
 			this.deploymentId = deploymentId;
 			this.instanceNumber = instanceNumber;
+			this.port = port;
 			builder.directory(workDir.toFile());
 			String workDirPath = workDir.toFile().getAbsolutePath();
 			this.stdout = Files.createFile(Paths.get(workDirPath, "stdout_" + instanceNumber + ".log")).toFile();
 			this.stderr = Files.createFile(Paths.get(workDirPath, "stderr_" + instanceNumber + ".log")).toFile();
 			builder.redirectOutput(this.stdout);
 			builder.redirectError(this.stderr);
 			builder.environment().put("INSTANCE_INDEX", Integer.toString(instanceNumber));
-			builder.environment().put("spring.application.index", Integer.toString(instanceNumber));
+			builder.environment().put("spring.application.index", Integer.toString(port));
```

Comment:
This may be oversight from my part but I change it because I think at some point we talked about this index needing to match guid? If so I can change it back as I don't think there is any meaning in metrics. Maybe it was these previous confusions with discussions for index vs. index vs. instance.